source/framework/tools/src/TRestStringHelper.cxx

```diff
@@ -638,6 +638,73 @@ string REST_StringHelper::ParameterNameToDataMemberName(string name) {
     }
 }
 
+/////////////////////////////////////////////
+/// \brief Reads fuction with parameter options from strig and returns it as TF1*. 
+///
+/// > Function as string following ROOT::TFormula conventions for parameters. 
+/// They are defined between square brackets and allow initial values, fixed values and ranges. 
+/// This has been created for fitting functions, where each parameter has its own restrictions.
+/// Examples: 
+/// -- Initial value: [0=3.5]
+/// -- Fixed value: [0==3.5]
+/// -- Range: [0=3.5(1,5)] The parameter 0 begin at 3.5 and it can move between 1 and 5.
+/// All parameters should be initialized.
+///
+/// Input: String with function and parameter options.
+///        Two doubles with the range of the function.
+/// Output: TF1* with the interpreted fuction. It contains all restrictions, ranges, etc.
+
+TF1* REST_StringHelper::ExtractTF1FromString(std::string func, double init, double end) {
+    string tf1 = func;
+    // Number of parameters
+    size_t n = std::count(func.begin(), func.end(), '[');
+    // Reading options 
+    int a = 0;
+    int optPos[n];
+    std::vector<std::string> options (n); // Vector of strings of any size.
+    for(int i = 0; i < n; i++){
+        optPos[i]=func.find("[", a); 
+        options[i]=func.substr(func.find("[", a)+1, func.find("]", func.find("[", a))-func.find("[", a)-1 ); 
+        a = func.find("[", a) + 1;
+    }
+    // Removing options from function string
+    for(int i = 0; i < n; i++){tf1.replace(optPos[n-1-i]+1, (func.find("]", optPos[n-1-i])-optPos[n-1-i]-1), std::string(1, func[optPos[n-1-i]+1]) );}
+    
+    // Function
+    const char * tf1c = tf1.c_str();
+    TF1* f = new TF1("f",tf1c,init,end);
```

Comment:
> Here, when the new creates the TF1 object, no delete is found. So, each time you call this method you create a new object in memory that is never freed.

I'm not sure that's actually true in case of ROOT. According to:

https://root.cern.ch/root/htmldoc/guides/users-guide/ObjectOwnership.html#ownership-by-the-master-troot-object-groot

the global `gROOT` object keeps track of `TF1` pointers. All this stuff is so opaque in ROOT that it's imo a PITA to deal with. One of the biggest reasons I don't want to deal with it personally.

Reply:
Right, thats why we the best is to explicitly keep `TF1` as a member of the class (non stored, i.e. `TF1 *fTF1Pointer; //!`) , then do the delete at each method call.

A standard solution is to keep the pointer at the header of the class, then at the method call one does.

```
if( !fTF1Pointer ) { delete fTF1Pointer; fTF1Pointer = nullptr; }
fTF1Pointer = new ...
```

But my question here is coming because `REST_StringHelper` is not a class itself, so it cannot encapsulate the pointer to be only in the context of the class. So, what would be a solution here to avoid for a memory leak? Any ideas? @nkx111 @juanangp

Reply:
We can leave in documentation that this method returns a `newed` object that needs manual deleting after calling. Just like `TRestMetadata::StringToElement()`. It would be the simplist way:

```
TF1* ff = ExtractTF1FromString(xxx);
// ff->DoSomething();
delete ff;
```

If we really want to get a safe object from the method, without concerning about deleting, maybe returning a std::shared_ptr is one of the solution.

> So, a standard solution is then to keep the pointer at the header of the class, then at the method call one does.

The approach of keeping a global ptr is not good. It may cause problems in multi-thread environment.

Reply:
> Perhaps, the TF1 object should be created externally, and this method would receive the TF1 function by reference, so that it can be modified inside. Then, perhaps this method would be better named InitializeTF1FromString.

I like this option, actually this function was conceived exactly for this so most of the times you end up with two functions in memory, the one created by ExtractTF1FromString and the one you define outside, something like  
`f = ExtractTF1FromString(fFunction, fFunctionRange.X(), fFunctionRange.Y());`. 
With InitializeTF1FromString we would avoid that and the function would work as I had originally in mind.

Reply:
Ok, I have renamed the method to `CreateTF1FromString`. In fact, this works as if it would be a ROOT::TF1 constructor, but without the new. I have added some documentation.

If you want to switch to `shared_ptr` or other solution, I will learn how it works, I am not familiar with.

I have updated rawlib according to that changes, deleting the TF1 function inside `ProcessEvent`. Created the a new PR rest-for-physics/rawlib#29.